src/browser/auth/Login.react.js

```diff
@@ -14,6 +14,10 @@ export default class Login extends Component {
     msg: PropTypes.object.isRequired
   };
 
+  static contextTypes = {
```

Comment:
I was thinking about extracting this to a Higher order component (as it was previously) but – context is now a documented feature of React and I find it more understandable this way. Especially since Router docs now use it this way.